StringUtils.c

```diff
@@ -140,3 +143,37 @@ char* String_readLine(FILE* fd) {
       at = buffer + bufSize - step;
    }
 }
+
+int xAsprintf(char** strp, const char* fmt, ...) {
+   va_list vl;
+   va_start(vl, fmt);
+   int _r = vasprintf(strp, fmt, vl);
+   va_end(vl);
+
+   if (_r < 0) {
+      fail();
+   }
+
+   return _r;
+}
+
+int xSnprintf(char* buf, int len, const char* fmt, ...) {
+   va_list vl;
+   va_start(vl, fmt);
+   int _n=vsnprintf(buf, len, fmt, vl);
```

Comment:
Please remove the leading underscore (and add spaces around `=`). (Originally `xAsprintf` was a macro, so it was about avoiding shadowing)